forloop_modules/function_handlers/database_handlers.py

```diff
@@ -424,15 +449,39 @@ def execute(self, node_detail_form):
         # else:
         #     raise HTTPException(status_code=response.status_code, detail="Error requesting new node from api")
         
-    def direct_execute(self, db_name, dbtable_name, selected_columns, column_name, operator, value, limit, new_var_name):        
-        db_table = get_db_table_from_db(table_name=dbtable_name, db_name=db_name)
-        db_instance = db_table.db1
+    def direct_execute(self, db_name, db_table_name, select, where_column_name, where_operator, where_value, limit, new_var_name):  
+        project_databases = get_all_databases_in_project()
+        db_dict = filter_database_by_name_from_all_project_databases(project_databases=project_databases, db_name=db_name)
+        
+        if db_dict is None:
+            # User selects from stored DBs so this shouldn't happen. If this is raised, these is an issue in code probably.
+            raise Exception(f'{self.icon_type}: No DB named {db_name} found in project DBs.')
+        
+        redis_key = create_redis_key_for_project_db_private_key(project_uid=aet.project_uid)
+        private_key_base64 = kv_redis.get(redis_key)
+        
+        if private_key_base64 is not None:
+            private_key = convert_base64_private_key_to_rsa_private_key(private_key_base64=private_key_base64)
```

Comment:
spaghetti, please wrap pk operation to a subfunction

Reply:
Do you mean the whole password decryption or just those 4 lines of retrieve pk from redis and parsing it? @DovaX